src/rgw/rgw_common.h

```diff
@@ -799,6 +806,7 @@ struct RGWBucketInfo
      ::encode(quota, bl);
      ::encode(num_shards, bl);
      ::encode(bucket_index_shard_hash_type, bl);
+     ::encode((uint32_t)index_type, bl);
```

Comment:
need to encode the following (with default values) before:
     ::encode(bucket_index_shard_hash_type, bl);
     ::encode(requester_pays, bl);
     ::encode(owner.tenant, bl);
     ::encode(has_website, bl);
     if (has_website) {
       ::encode(website_conf, bl);
     }
     ::encode((uint32_t)index_type, bl);